### PR TITLE
ISPN-7254 Accessing content without needed permissions should display error message

### PR DESCRIPTION
**Pull request by vblagoje**
Master only. Error dialog is raised. Upon clicking OK admin is taken to containers page. See https://issues.jboss.org/browse/ISPN-7254 for more details

### CODE REVIEW

File: src/services/schemas/SchemasService.ts [hunk 1]
```diff
@@ -25,7 +25,7 @@ export class SchemaService {
     let deferred: ng.IDeferred<ISchemaDefinition> = this.$q.defer<ISchemaDefinition>();
     this.findTarget(container.serverGroup)
       .then(coordinator => {
-          return this.dmrService.executePost({
```
Comment (ryanemerson) [inline src/services/schemas/SchemasService.ts]: We need this return, as this means that the result of the promise `this.dmrService.executePost(...` is passed to L35's then clause. It does not mean that the overarching function `getProtoSchema` returns this result.

File: src/services/schemas/SchemasService.ts [hunk 2]
```diff
@@ -68,7 +71,7 @@ export class SchemaService {
     let deferred: ng.IDeferred<void> = this.$q.defer<void>();
     this.findTarget(container.serverGroup)
       .then(coordinator => {
-        return this.dmrService.executePost({
+        this.dmrService.executePost({
```
Comment (ryanemerson) [inline src/services/schemas/SchemasService.ts]: We need this return as it ensures that the result of the executePost is passed to the the chained `then.` call

File: src/services/schemas/SchemasService.ts [hunk 3]
```diff
@@ -39,12 +39,15 @@ export class SchemaService {
...
           operation: "get-proto-schema-names"
-        }).then(names => deferred.resolve(names));
+        }).then(
+          (names) => deferred.resolve(names),
+          (error) => deferred.reject(error));
       },
-      error => deferred.reject(error));
+      (error) => deferred.reject(error)
+    );
     return deferred.promise;
```
Comment (ryanemerson) [inline src/services/schemas/SchemasService.ts]: The original code is definitely wrong, but I believe that the second `.then` should be chained after the first, see below:

```typescript
registerProtoSchema(container: ICacheContainer, schema: ISchemaDefinition): ng.IPromise<void> {
    let deferred: ng.IDeferred<void> = this.$q.defer<void>();
    this.findTarget(container.serverGroup)
      .then(coordinator => {
          return this.dmrService.executePost({
            address: this.getContainerAddress(container.name, coordinator),
            operation: "register-proto-schemas",
            "file-names": [schema.fileName],
            "file-contents": [schema.fileContents]
          });
        },
        error => deferred.reject(error))
      .then(() => deferred.resolve(), error => deferred.reject(error));
    return deferred.promise;
  }
```

File: src/ManagementConsole.ts [hunk 4]
```diff
@@ -142,10 +144,12 @@ module.run(($templateCache: ITemplateCacheService) => {
 
 // TODO remove if only used for debugging
```
Comment (ryanemerson) [inline src/ManagementConsole.ts]: Lets remove this TODO

File: src/module/cache-container/config/ContainerConfig.ts [hunk 5]
```diff
@@ -11,6 +11,7 @@ import {ContainerConfigService} from "../../../services/container-config/Contain
 import {TasksCtrl} from "./tasks/TasksCtrl";
 import {SecurityCtrl} from "./security/SecurityCtrl";
 import IQService = angular.IQService;
+import {ContainerService} from "../../../services/container/ContainerService";
```
Comment (ryanemerson) [inline src/module/cache-container/config/ContainerConfig.ts]: Unnecessary import added.

File: src/services/schemas/SchemasService.ts (hunk 2 above)
Comment (vblagoje) [inline src/services/schemas/SchemasService.ts]: +1

**Comment by vblagoje** [comment]
@ryanemerson updated. Please check again and if all good integrate and issue PR for the product side as well

**Comment by ryanemerson** [comment]
Integrated cf411575cc5db0586890867fd473a4c3d77708f7, thanks @vblagoje

**Pull request closed without merging**